public/static/app/perf/perf.js

```diff
@@ -903,18 +907,53 @@ $http.get(templateUrl).success(function(template) {
     }
     return data;
   };
-}).factory('loadWhitelist', function ($q, PointsDataService, WhitelistDataService) {
+}).factory('getVisibleRange', function () {
+  return function (trend_data) {
+    const {
+      legacy = [], cedar = []
+    } = trend_data;
+    let from = null;
+    let to = null;
+
+    const dates = _.chain([].concat(legacy || [], cedar || []))
+                    .pluck("create_time")
+                    .uniq()
+                    .sortBy((timestamp) => new Date(timestamp))
+                    .value();
+    if(!_.isEmpty(dates)) {
+      from = dates[0];
+      if (from !== dates[dates.length-1]) {
```

Comment:
what does the dates data structure look like? is this checking if there is only 1 entry in the dates array?

Reply:
The chain command gets all the ```create_times``` values from both legacy / cedar points and sorts them into a uniq list.

So the data structure would be ```["timestamp 1", .... "timestamp n"]```.

```!_.isEmpty``` ensures that there is at least one element and ```dates[0] !== dates[dates.length-1]``` if there is more than one element.